db/cache.go

```diff
@@ -0,0 +1,85 @@
+package db
+
+import (
+	"database/sql"
+)
+
+// CacheRows is a container to store the values from sql.Rows
+type CacheRows struct {
+	rows  [][]interface{}
+	index int
+	err   error
+}
+
+// Reset resets the index to 0 - must be called before iteration
+func (c *CacheRows) Reset() {
+	c.index = 0
+}
+
+// Next iterator
+func (c *CacheRows) Next() bool {
+	c.index++
+	return c.index <= len(c.rows)
+}
+
+// Err returns the error, if any, encountered in the iteration
+func (c *CacheRows) Err() error {
+	return c.err
+}
+
+// Close does nothing, but it's here for backawards compatibility with sql.Rows
+func (c *CacheRows) Close() error {
+	return c.err
+}
+
+// Scan scanner - similar to sql.Rows.Scan, but you must pass in a Scanner type such as: sql.NullString
+func (c *CacheRows) Scan(dest ...sql.Scanner) error {
+	row := c.rows[c.index-1]
+	for i, scanner := range dest {
+		err := scanner.Scan(row[i])
+		if err != nil {
+			c.err = err
+			return err
+		}
+	}
+	return nil
+}
+
+// Fill fill with rows
```

Comment:
Maybe have a NewCacheRows(rows *sql.Rows) (*CacheRows, error) constructor instead of having a Fill method? Since you always need to call Fill before use, it would make the interface simpler.  Also Fill should call rows.Close() and rows.Err()

Reply:
agreed with this.